concent_api/core/models.py

```diff
@@ -73,3 +75,216 @@ class MessageAuth(Model):
+
+    # Relation to concent client in provider context.
+    provider                    = models.ForeignKey(Client, related_name = 'subtasks_as_provider')
+
+    state                       = models.CharField(max_length = 32, choices = SubtaskState.choices())
+
+    # If in an active state, it's the time at which Concent automatically transitions to a passive state if it does
+    # not get the information it expects from the client (a timeout). Must be NULL in a passive state.
+    next_deadline               = models.DateTimeField(null = True)
+
+    # Related messages
+    task_to_compute             = models.ForeignKey(StoredMessage, null = True)
+    report_computed_task        = models.ForeignKey(StoredMessage, null = True)
+    ack_report_computed_task    = models.ForeignKey(StoredMessage, null = True)
+    reject_report_computed_task = models.ForeignKey(StoredMessage, null = True)
+    subtask_results_accepted    = models.ForeignKey(StoredMessage, null = True)
+    subtask_results_rejected    = models.ForeignKey(StoredMessage, null = True)
+
+
+class ReceiveQueue(models.Model):
+    """
+    Stores information about messages to be returned from the `receive` or `receive-out-of-band` endpoint.
+    """
+
+    class ConcentResponseType(ChoiceEnum):
+        ForceReportComputedTask         = 'ForceReportComputedTask'
+        ForceReportComputedTaskResponse = 'ForceReportComputedTaskResponse'
+        VerdictReportComputedTask       = 'VerdictReportComputedTask'
+        ForceGetTaskResultRejected      = 'ForceGetTaskResultRejected'
+        ForceGetTaskResultFailed        = 'ForceGetTaskResultFailed'
+        ForceGetTaskResultUpload        = 'ForceGetTaskResultUpload'
+        ForgetGetTaskResultDownload     = 'ForgetGetTaskResultDownload'
+        ForceSubtaskResults             = 'ForceSubtaskResults'
+        SubtaskResultsSettled           = 'SubtaskResultsSettled'
+        ForceSubtaskResultsResponse     = 'ForceSubtaskResultsResponse'
+        SubtaskResultsRejected          = 'SubtaskResultsRejected'
+
+    class ReceiveQueueType(ChoiceEnum):
+        Receive             = 'receive'
+        ReceiveOutOfBand    = 'receive_out_of_band'
+
+    response_type        = models.CharField(max_length = 32, choices = ConcentResponseType.choices())
+    client               = models.ForeignKey(Client)
+    queue                = models.CharField(max_length = 32, choices = ReceiveQueueType.choices())
+
+    # TRUE if the client has already fetched the message.
+    delivered            = models.BooleanField(default = False)
+
+    subtask              = models.ForeignKey(Subtask)
+    created_at           = models.DateTimeField(default = get_current_utc_timestamp)
+
+
+class AbstractStoredMessage(models.Model):
+
+    timestamp = models.DateTimeField()
+    signature = Base64Field(max_length = 64)
+
+    class Meta:
+        abstract = True
+
+
+class StoredTaskToCompute(AbstractStoredMessage):
```

Comment:
Don't create task specific tables yet. This is a separate task (#141). It's separate because it's something that's can be done after the release if necessary.